### PR TITLE
Add sign-in/up pages and reducer for auth

### PR DESCRIPTION
**Pull request by anthonytate**
Base pages and functionality only. Will add CSS files later

### CODE REVIEW

File: client/src/authentication/auth-slice.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      signOutUser();
+      state.loggedIn = false;
+    },
+    setError(state, action) {
+      state.error = action.payload;
+    },
+  },
+});
+
+export const {authenticate, deauthenticate, saveUserId, setError} =
```
Comment (hack-heart) [inline client/src/authentication/auth-slice.js:31]: PLEASE FIX: saveUserId is not a defined action

File: client/src/pages/sign-up.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        />
+        <input type='password' name='password' placeholder='password' onChange={(event) => { setPassword(event.target.value); }}/>
+        <button
+          type="button"
+          onClick={() => { props.authenticate({email, password, newUser: true}); }}
+        >
+          Submit
+        </button>
+      </form>
+      <br />
```
Comment (hack-heart) [inline client/src/pages/sign-up.js:28]: nit: I forgot to mention this but the <br> tag does not appear necessary here

File: client/src/pages/sign-in.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        />
+        <input type='password' name='password' placeholder='password' onChange={(event) => { setPassword(event.target.value); }}/>
+        <button
+          type="button"
+          onClick={() => { props.authenticate({email, password, newUser: false}); }}
+        >
+          Submit
+        </button>
+      </form>
+      <br />
```
Comment (hack-heart) [inline client/src/pages/sign-in.js:28]: nit: br tag not necessary

File: client/src/api/user-api.js [hunk 4]
```diff
@@ -13,23 +13,22 @@ const storeUserInDatabase = (id) => {
...
 export const signInUser = (email, password) => {
   firebase.auth()
-    .signInWithEmailAndPassword(email, password)
-    .then(() => firebase.auth().currentUser.uid)
-    .then((uid) => uid)
-    .catch((error) => error);
+      .signInWithEmailAndPassword(email, password)
+      .then(() => firebase.auth().currentUser.uid)
+      .then((uid) => uid)
+      .catch((error) => error);
```
Comment (MrM117) [inline client/src/api/user-api.js:29]: nit: same here

**Pull request merged**